src/vs/workbench/contrib/files/browser/editors/fileEditorInput.ts

```diff
@@ -421,6 +421,10 @@ export class FileEditorInput extends AbstractTextResourceEditorInput implements
 	}
 
 	override matches(otherInput: EditorInput | IUntypedEditorInput): boolean {
+		if (isResourceMergeEditorInput(otherInput)) {
```

Comment:
@lramos15 yes, this is indeed a bug in the `matches` implementation but should this not be tackled where it actually happens:

https://github.com/microsoft/vscode/blob/0dddb856fd89d55b1dadc97ab49aca800f3eb8ff/src/vs/workbench/common/editor/editorInput.ts#L273

Because in theory it could impact other editors calling `super.matches()` as well, no?

Reply:
Doing a quick pass, the following editor inputs declare an `editorId` of `default` and thus would wrongly `match` on a merge editor input:
* [ ] [`TextResourceEditorInput`](https://github.com/microsoft/vscode/blob/0dddb856fd89d55b1dadc97ab49aca800f3eb8ff/src/vs/workbench/common/editor/textResourceEditorInput.ts#L87-L89)
* [ ] [`FileEditorInput`](https://github.com/microsoft/vscode/blob/0dddb856fd89d55b1dadc97ab49aca800f3eb8ff/src/vs/workbench/contrib/files/browser/editors/fileEditorInput.ts#L42-L44)
* [x] [`MergeEditorInput`](https://github.com/microsoft/vscode/blob/0dddb856fd89d55b1dadc97ab49aca800f3eb8ff/src/vs/workbench/contrib/mergeEditor/browser/mergeEditorInput.ts#L89-L91)
* [ ] [`UntitledTextEditorInput`](https://github.com/microsoft/vscode/blob/0dddb856fd89d55b1dadc97ab49aca800f3eb8ff/src/vs/workbench/services/untitled/common/untitledTextEditorInput.ts#L31-L33)

I actually think the merge editor input is doing it in a good way, so your fix should then maybe expand to the other inputs:

https://github.com/microsoft/vscode/blob/0dddb856fd89d55b1dadc97ab49aca800f3eb8ff/src/vs/workbench/contrib/mergeEditor/browser/mergeEditorInput.ts#L180-L188

Reply:
Makes sense. I think the idea here is to move away from calling `super.matches` in a lot of places as it is a basic implementation that doesn't really fully take into consideration many of the editor specifics. I'll try to tackle some of these in a separate PR

Reply:
Yeah makes sense 👍

Reply:
Can one of you please approve this PR then?